feat!: Address descriptors

tcliff111 opened the pull request:
Thank you for opening a Pull Request!

---

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x ] Make sure to open a GitHub issue as a bug/feature request before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ x] Ensure the tests and linter pass
- [x ] Code coverage does not decrease (if any source code was changed)
- [ x] Appropriate docs were updated (if necessary)

Fixes #294 🦕

Commit:
feat! Added Address Descriptors to Geocoding response. Refactored Geocoding response to allow fields outside the geocoding result to be exposed through the client.

Commit:
feat! Added Address Descriptors to Geocoding response. Refactored Geocoding response to allow fields outside the geocoding result to be exposed through the client.

Commit:
feat! Added Address Descriptors to Geocoding response. Refactored Geocoding response to allow fields outside the geocoding result to be exposed through the client.

Commit:
feat! Added Address Descriptors to Geocoding response. Refactored Geocoding response to allow fields outside the geocoding result to be exposed through the client.

Commit:
feat! Added Address Descriptors to Geocoding response. Refactored Geocoding response to allow fields outside the geocoding result to be exposed through the client.

wangela reviewed (changes requested):
- Let's update the README "Usage" section with either a demonstration of Geocoding or a pointer to see examples by examining tests.
- Does https://github.com/googlemaps/google-maps-services-go/blob/master/examples/geocoding/cmdline/main.go need to be updated as well?

tcliff111 commented:
> * Let's update the README "Usage" section with either a demonstration of Geocoding or a pointer to see examples by examining tests.
> * Does https://github.com/googlemaps/google-maps-services-go/blob/master/examples/geocoding/cmdline/main.go need to be updated as well?

Just updated the README and also updated the main.go file.

wangela commented on README.md:
````diff
@@ -90,6 +90,37 @@ func main() {
 }
 ```
 
+Sample usage of the Geocoding API with an API key to get an [Address Descriptor](https://mapsplatform.google.com/demos/address-descriptors/):
````
https://developers.google.com/maps/documentation/geocoding/address-descriptors/requests-address-descriptors might be a more useful destination for the hyperlink

tcliff111 committed:
Update README.md
